pkg/option/config.go

```diff
@@ -823,6 +829,203 @@ const (
+		Flags: []string{
+			FQDNRejectResponseCode,
+			ToFQDNsEnablePoller,
+			ToFQDNsEnablePollerEvents,
+			ToFQDNsMaxIPsPerHost,
+			ToFQDNsMinTTL,
+			ToFQDNsPreCache,
+			ToFQDNsProxyPort,
+			FQDNProxyResponseMaxDelay,
+			ToFQDNsEnableDNSCompression,
+			ToFQDNsMaxDeferredConnectionDeletes,
+		},
+	},
+	{
+		// K8S flags section for the cilium-agent help template
+		Name: "K8S flags",
+		Flags: []string{
+			K8sAPIServer,
+			K8sKubeConfigPath,
+			K8sNamespaceName,
+			K8sRequireIPv4PodCIDRName,
+			K8sRequireIPv6PodCIDRName,
+			K8sWatcherEndpointSelector,
+			K8sWatcherQueueSize,
+			K8sEventHandover,
+			DisableK8sServices,
+			AnnotateK8sNode,
+			K8sForceJSONPatch,
+			DisableCiliumEndpointCRDName,
+			K8sHeartbeatTimeout,
+			EnableExternalIPs,
+			K8sEnableEndpointSlice,
+		},
+	},
+	{
+		// Cluster flags section for the cilium-agent help template
+		Name: "Cluster flags",
+		Flags: []string{
+			ClusterIDName,
+			ClusterName,
+			ClusterMeshConfigName,
+		},
+	},
+	{
+		// Route flags section for the cilium-agent help template
+		Name: "Route flags",
+		Flags: []string{
+			SingleClusterRouteName,
+			EnableEndpointRoutes,
+			EnableLocalNodeRoute,
+			BlacklistConflictingRoutes,
+			EnableAutoDirectRoutingName,
+		},
+	},
+	{
+		// HTTP flags section for the cilium-agent help template
+		Name: "HTTP flags",
+		Flags: []string{
+			HTTPIdleTimeout,
+			HTTPMaxGRPCTimeout,
+			HTTPRequestTimeout,
+			HTTPRetryCount,
+			HTTPRetryTimeout,
+		},
+	},
+	{
+		// Debug, logging and trace flags section for the cilium-agent help template
+		Name: "Debug, logging and trace flags",
+		Flags: []string{
+			DebugArg,
+			DebugVerbose,
+			EnableTracing,
+			LogDriver,
+			LogOpt,
+			LogSystemLoadConfigName,
+			EnvoyLog,
+			EnableEndpointHealthChecking,
+			EnableHealthChecking,
+		},
+	},
+	{
+		// Metrics and monitoring flags section for the cilium-agent help template
+		Name: "Metrics and monitoring flags",
+		Flags: []string{
+			Metrics,
+			MonitorAggregationName,
+			MonitorAggregationFlags,
+			MonitorAggregationInterval,
+			MonitorQueueSizeName,
+			PrometheusServeAddr,
+		},
+	},
+	{
+		// IP flags section for the cilium-agent help template
+		Name: "IP flags",
+		Flags: []string{
+			EnableIPv4Name,
+			EnableIPv6Name,
+			IPAllocationTimeout,
+			IPAM,
+			IPv4ClusterCIDRMaskSize,
+			IPv4NodeAddr,
+			IPv6NodeAddr,
+			IPv4PodSubnets,
+			IPv6PodSubnets,
+			IPv4ServiceRange,
+			IPv6ServiceRange,
+			IPv4Range,
+			IPv6Range,
+			LoopbackIPv4,
+			IPv6ClusterAllocCIDRName,
+			MTUName,
+			NAT46Range,
+		},
+	},
+	{
+		// KVstore flags section for the cilium-agent help template
+		Name: "KVstore flags",
+		Flags: []string{
+			KVStore,
+			KVStoreOpt,
+			KVstoreConnectivityTimeout,
+			KVstorePeriodicSync,
+		},
+	},
+	{
+		// Encrpytion flags section for the cilium-agent help template
+		Name: "Encryption flags",
+		Flags: []string{
+			EncryptInterface,
+			EncryptNode,
+			EnableHostReachableServices,
```

Comment:
`EnableHostReachableServices` isn't encryption either.

Reply:
will move it.